cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/state/AppState.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.internal.app.store.state;
+
+import javax.annotation.Nullable;
+
+/**
+ * Represents an application state.
+ */
+public class AppState {
+  private final String namespace;
```

Comment:
use NamespaceId instead of a String. This ensures that it is a valid namespace (in terms of the character set)